### ISSUE TITLE
Refactor Device Switcher animation

### ISSUE THREAD

**Issue by vytick** [trezor/trezor-suite#12541]
Refactor device switcher open/close animation in style of accordion https://github.com/trezor/trezor-suite/pull/12534 to make code more sustainable

**Comment by vytick** [comment]
I've briefly tried to refactor and still run into issue with measuring the component height. It still kept returning 0 for height.

**Comment by matejkriz** [comment]
Now it's not responsive at all so different font size breaks it 
<img width="349" alt="image" src="https://github.com/user-attachments/assets/1bae6c9f-571f-47f5-9f06-7171df9a6752">

**Comment by matejkriz** [comment]
Wasn’t this resolved by https://github.com/trezor/trezor-suite/pull/20561 @BrantalikP ?

**Comment by BrantalikP** [comment]
yep, that one has been fixed along https://github.com/trezor/trezor-suite/pull/20561

**Comment by BrantalikP** [comment]
I briefly recoll that changing the wrapper from `Fragment` to `Box` did the [trick](https://github.com/trezor/trezor-suite/pull/20561/files#diff-1f193c62cbc357f695fb735d13881a111b38ee6332bd688c199dbdfd115c4169R41)
